src/components/checkbox/checkbox.spec.js

```diff
@@ -246,5 +247,45 @@ describe('mdCheckbox', function() {
       expect(isChecked(checkbox)).toBe(false);
       expect(checkbox.hasClass('ng-invalid')).toBe(true);
     });
+
+    describe('with the md-indeterminant attribute', function() {
+
+      it('should set md-indeterminant attr to true by default', function() {
+        var checkbox = compileAndLink('<md-checkbox md-indeterminant></md-checkbox>');
+
+        expect(checkbox).toHaveClass(INDETERMINATE_CSS);
+      });
+
+      it('should set its HTMLInputElement.indeterminate to be undefined when not md-indeterminant', function() {
+        var checkbox = compileAndLink('<md-checkbox md-indeterminant="false"></md-checkbox>');
+
+        expect(checkbox[0].indeterminate).toBe(undefined);
+      });
+
+      it('should set its HTMLInputElement.indeterminate to be "true" when md-indeterminant', function() {
+        var checkbox = compileAndLink('<md-checkbox md-indeterminant></md-checkbox>');
+
+        expect(checkbox[0].indeterminate).toBe(true);
+      });
+
+      it('should be set "md-indeterminant" class according to a passed in function', function() {
+        pageScope.isIndeterminate = function() { return true; } ;
+
```

Comment:
@DerekLouie Maybe you can also add a test, which changes the `indeterminate` expression on runtime as well?